mypy/errorcodes.py

```diff
@@ -67,6 +67,9 @@ def __str__(self) -> str:
 TYPEDDICT_ITEM: Final = ErrorCode(
     "typeddict-item", "Check items when constructing TypedDict", "General"
 )
+TYPPEDICT_UNKNOWN_KEY: Final = ErrorCode(
+    "typeddict-unknown-key", "Check unknown keys when constructing TypedDict", "General"
+)
```

Comment:
As mentioned on the issue, for consistency this error code should be also used for errors in case like:
```python
d["unknown"]
d["unknown"] = 42
```

Reply:
OK, reading an unknown key may be too much, but setting it (i.e. `d["unknown"] = 42`) should definitely use the same error code. I don't see a difference between:
```python
d: TD = {"known": 1, "unknown": 2}
```
and
```python
d: TD = {"known": 1}
d["unknown"] = 2
```

Reply:
I think that's very reasonable. Will ping you when I have that sorted.

Reply:
Done!

Not sure if the solution is pretty. Added a test for it on the typeddict file.